src/convert.rs

```diff
@@ -167,37 +176,8 @@ pub fn elf_to_tbf(
         false
     }
 
-    /// Helper function to find the address of the first section inside a given
-    /// segment.
-    ///
-    /// This is necessary because the flash segment is not guaranteed
-    /// to start at the same address as the first section.
-    fn find_first_section_address_in_segment<'a>(
-        input: &'a elf::File,
-        segment: &elf::types::ProgramHeader,
-    ) -> Option<u32> {
-        let segment_start = segment.offset as u32;
-        let segment_size = segment.filesz as u32;
-        let segment_end = segment_start + segment_size;
-
-        let mut first_section_address: Option<u32> = None;
-        for section in input.sections.iter() {
-            let section_start = section.shdr.offset as u32;
-            let section_size = section.shdr.size as u32;
-            let section_end = section_start + section_size;
-
-            if section_start >= segment_start && section_end <= segment_end && section_size > 0 {
-                first_section_address = match first_section_address {
-                    Some(first_address) => Some(cmp::min(first_address, section.shdr.addr as u32)),
-                    None => Some(section.shdr.addr as u32),
-                };
-            }
-        }
-        first_section_address
-    }
-
     // Do flash address.
-    for segment in &input.phdrs {
+    for segment in &elf_file.phdrs {
```

Comment:
Since we are already looping segments below, can we merge the logic in this loop with the one below?

Reply:
My understanding is that the first loop is required to fetch some address in order to build the tbf header. I don't think we could make it in a single loop. Can you clarify?